options/options_parser.cc

```diff
@@ -656,8 +660,17 @@ Status RocksDBOptionsParser::VerifyRocksDBOptionsFromFile(
     const std::vector<ColumnFamilyOptions>& cf_opts,
     const std::string& file_name, FileSystem* fs,
     OptionsSanityCheckLevel sanity_check_level, bool ignore_unknown_options) {
+  // We infer option file readhead size from log readahead size.
+  // If it is not given, use 512KB.
+  size_t file_readahead_size = db_opt.log_readahead_size;
+  if (file_readahead_size == 0) {
+    const size_t kDefaultOptionFileReadAheadSize = 512 * 1024;
```

Comment:
Maybe check compaction_readahead_size too? For spinning disks, 1-2MB may be more appropriate.

Reply:
Let me make it 1MB. I'm a little bit concern about DRAM usage. option files tend to be very small and rarely take more than a few hundred KBs while right now for readahead we allocate the whole buffer size.

Reply:
If the larger option files are usually < 512KB and can be read in one shot, then 512KB is probably ok. In any case, this is a vast improvement over the previous implementation.